CMakeLists.txt

```diff
@@ -267,12 +261,18 @@ if(AMENT_ENABLE_TESTING)
   ament_lint_auto_find_test_dependencies()
 endif()
 
-  file(
-    COPY "${PROJECT_NAME}"
-    DESTINATION "${CMAKE_CURRENT_BINARY_DIR}"
-  )
-
-ament_python_install_package(${PROJECT_NAME} PACKAGE_DIR "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}")
+# Copy the Python code to the folder where the C extensions will be generated.
+# This creates a folder in ${CMAKE_CURRENT_BINARY_DIR}/rclpy which has all the
+# necessary Python code and C Python extensions for running tests.
+# This folder is also ready to be installed directly as a Python package.
+file(
+  COPY "${PROJECT_NAME}"
+  DESTINATION "${CMAKE_CURRENT_BINARY_DIR}"
+)
```

Comment:
I still don't see why this should be done. Copying files like this should be avoided at all cost. Can someone argue why this is absolutely necessary?